pandas/io/parsers.py

```diff
@@ -2354,12 +2354,16 @@ def _set_no_thousands_columns(self):
         # Create a set of column ids that are not to be stripped of thousands
         # operators.
         noconvert_columns = set()
+        if self._col_indices is not None:
```

Comment:
sort _col_indices when its created. I would try actually just setting it to the list(range(len(self.columns))) otherwise when its set (rather than doing it here).

Reply:
Moved it up and sorted when set immediately